File: article-api/src/main/kotlin/com/sns/article/reaction/ui/ReactionController.java
package com.sns.article.reaction.ui;

import java.net.URI;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.sns.article.reaction.application.ReactionService;
import com.sns.article.reaction.dto.ReactionDto;

// TODO : 권한 체크
@RestController
@RequestMapping("/api/v1/reactions")
public class ReactionController {
	private final ReactionService reactionService;

	public ReactionController(ReactionService reactionService) {
		this.reactionService = reactionService;
	}

	@PostMapping
	public ResponseEntity<Void> createReaction(@RequestBody ReactionDto reactionDto) {
		ReactionDto reaction = reactionService.create(reactionDto);
		return ResponseEntity.created(URI.create("/api/v1/reactions" + reaction.getId())).build();
	}

	@GetMapping("/{reactionId}")
	public ResponseEntity<ReactionDto> findReaction(@PathVariable Long reactionId) {
		ReactionDto reaction = reactionService.findById(reactionId);
		return ResponseEntity.ok(reaction);
	}

	@DeleteMapping("/{reactionId}")
	public ResponseEntity<Void> deleteReaction(@PathVariable Long reactionId) {
		reactionService.delete(reactionId);
		return ResponseEntity.noContent().build();
	}
}

File: article-api/src/main/kotlin/com/sns/article/reaction/ui/ReactionController.kt
package com.sns.article.reaction.ui

import com.sns.article.reaction.application.ReactionService
import com.sns.article.reaction.dto.ReactionDto
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.net.URI

// TODO : 권한 체크
@RestController
@RequestMapping("/api/v1/reactions")
class ReactionController(private val reactionService: ReactionService) {
    @PostMapping
    fun createReaction(@RequestBody reactionDto: ReactionDto): ResponseEntity<Void> {
        val reaction = reactionService.create(reactionDto)
        return ResponseEntity.created(URI.create("/api/v1/reactions/" + reaction.id)).build()
    }

    @GetMapping("/{reactionId}")
    fun findReaction(@PathVariable reactionId: Long): ResponseEntity<ReactionDto> {
        val reaction = reactionService.findById(reactionId)
        return ResponseEntity.ok(reaction)
    }

    @DeleteMapping("/{reactionId}")
    fun deleteReaction(@PathVariable reactionId: Long): ResponseEntity<Void> {
        reactionService.delete(reactionId)
        return ResponseEntity.noContent().build()
    }
}
